### PR TITLE
SALTO-6919: fix convertJsonIdsToReferences

### PR DESCRIPTION
**Pull request by edenhassid**
fix convertJsonIdsToReferences to work when json looks like this:
"id":"\<number\>"

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/zendesk-adapter/src/filters/handle_template_expressions.ts [hunk 1]
```diff
@@ -416,8 +416,16 @@ const replaceFormulasWithTemplates = ({
       if (!ID_KEY_IN_JSON_REGEX.test(expression)) {
         return expression
       }
+      // "id": "29607168675603" or "id": 29607168675603
       const idRegex = /\d+/
-      const id = expression.match(idRegex)?.[0]
+      const match = expression.match(idRegex)
+      if (!match || !match.index) {
+        log.error(`Error parsing id in expression, could not find a match: ${expression}`)
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/filters/handle_template_expressions.ts:423]:
```suggestion
        log.warn(`Error parsing id in expression, could not find a match: ${expression}`)
```

File: packages/zendesk-adapter/src/filters/handle_template_expressions.ts [hunk 2]
```diff
@@ -416,8 +416,16 @@ const replaceFormulasWithTemplates = ({
...
+        return expression
+      }
+      const id = match[0]
+      const prefix = expression.slice(0, match.index)
+      const suffix = expression.slice(match.index + id?.length, expression.length)
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/filters/handle_template_expressions.ts:428]: weird that you need the nullable here, if it's really the case add a fallback

File: packages/zendesk-adapter/test/filters/handle_template_expressions.test.ts [hunk 3]
```diff
@@ -498,7 +498,7 @@ describe('handle templates filter', () => {
             parts: [
               `{\n\t"ticket": {\n\t\t"custom_fields": [\n\t\t\t{\n\t\t\t\t"id": ${placeholder3.value.id},\n\t\t\t\t"testdc": "{{`,
               new ReferenceExpression(dynamicContentRecord.elemID, dynamicContentRecord),
-              `}}"\n\t\t\t}\n\t\t],\n\t\t"id": ${placeholder2.value.id}\n\t},\n\t"id": ${placeholder3.value.id}\n}\n`,
+              `}}"\n\t\t\t}\n\t\t],\n\t\t"id": ${placeholder2.value.id}\n\t},\n\t"id": "${placeholder3.value.id}"\n}\n`,
```
Comment (adfineberg) [inline packages/zendesk-adapter/test/filters/handle_template_expressions.test.ts:501]: do you want to leave the previous case as well?

File: packages/zendesk-adapter/src/filters/handle_template_expressions.ts [hunk 4]
```diff
@@ -428,9 +436,8 @@ const replaceFormulasWithTemplates = ({
         return expression
       }
 
-      const expressionWithoutId = expression.replace(idRegex, '')
       const idInstance = instance ?? createMissingInstance(ZENDESK, 'unknown', id)
-      return [expressionWithoutId, new ReferenceExpression(idInstance.elemID, idInstance)]
+      return [prefix, new ReferenceExpression(idInstance.elemID, idInstance), suffix ?? '']
```
Comment (adfineberg) [inline packages/zendesk-adapter/src/filters/handle_template_expressions.ts:440]: does the code compact the parts later? or disregard empty strings?

**Review by adfineberg — COMMENTED** [review]
Nice! A few small comments 👯

File: packages/zendesk-adapter/src/filters/handle_template_expressions.ts (hunk 4 above)
Comment (edenhassid) [reply packages/zendesk-adapter/src/filters/handle_template_expressions.ts:440]: yes we have `extractTempate` which calls `createTemplateExpression` which uses `compactTemplateParts`

File: packages/zendesk-adapter/test/filters/handle_template_expressions.test.ts (hunk 3 above)
Comment (edenhassid) [reply packages/zendesk-adapter/test/filters/handle_template_expressions.test.ts:501]: it have both cases here cases

**Review by adfineberg — APPROVED** [review]

**Pull request merged**